lib/pure/asynchttpserver.nim

```diff
@@ -295,15 +301,27 @@ proc processClient(server: AsyncHttpServer, client: AsyncSocket, address: string
   var request = newFutureVar[Request]("asynchttpserver.processClient")
   request.mget().url = initUri()
   request.mget().headers = newHttpHeaders()
+  request.mget().disableKeepalive = false
   var lineFut = newFutureVar[string]("asynchttpserver.processClient")
   lineFut.mget() = newStringOfCap(80)
 
+  let startTimeout = now()
   while not client.isClosed:
+    let fds = activeDescriptors()
+    # Disables the keepalive if the active file descriptors exceeds the maxFDs value
+    # or if the maximum keepalive timeout is exceeded. 
+    # The maxFDs should be replaced by the keepaliveConn?
+    if (fds > server.maxFDs) or ((now() - startTimeout).inSeconds > server.keepaliveTimeout):
+      request.mget().disableKeepalive = true
```

Comment:
No, you shouldn't do this at all. You should check the headers and see whether keep-alive has been turned off.

Reply:
@dom96 can you help me with the code? I want to close the connection when the number of file descriptors is high. Even if the value of keepalive is on I want to close the connection anyway. It is temporary until the number of files descriptors goes down, when the value goes down the connection can remain open. Please help me to solve the issue!

Reply:
@mrhdias This isn't what you should do though! Don't close connections when the number of FDs is high, this will just lead to more problems.